### PR TITLE
Code refactor, fixes and minor optimizations

### PR DESCRIPTION
**Pull request by GermanAizek**
Hi @Exzap, @Crementif,
This PR will be correcting the code for better readability, bringing it to the latest C++ standard, using some optimizations and fixing bugs.

### CODE REVIEW

File: src/Cafe/CafeSystem.cpp [hunk 1]
```diff
@@ -638,7 +638,7 @@ namespace CafeSystem
 		FSCDeviceHostFS_Mount(std::string("/internal/code/").c_str(), _pathToUtf8(executablePath.parent_path()), FSC_PRIORITY_BASE);
 		std::string internalExecutablePath = "/internal/code/";
 		internalExecutablePath.append(_pathToUtf8(executablePath.filename()));
-		_pathToExecutable = internalExecutablePath;
+		_pathToExecutable = std::move(internalExecutablePath);
```
Comment (Exzap) [inline src/Cafe/CafeSystem.cpp:641]: Needlessly verbose and may trip up someone in the future who tries to use either `rpxPath` or `internalExecutablePath`. This code runs once on game launch so it's not a worthy candidate for shaving of CPU microseconds anyway.

File: src/Cafe/GraphicPack/GraphicPack2PatchesApply.cpp [hunk 2]
```diff
@@ -125,14 +125,14 @@ double _cbResolveConstant(std::string_view varname)
...
 			std::string moduleName = importName.substr(0, idxDot);
-			std::string functionName = importName.substr(idxDot + 1);
 			uint32 rplHandle = RPLLoader_GetHandleByModuleName(moduleName.c_str());
 			if (rplHandle == RPL_INVALID_HANDLE)
 			{
 				importError = " (module not found)";
 			}
 			else
 			{
+				std::string functionName = importName.substr(idxDot + 1);
```
Comment (Exzap) [inline src/Cafe/GraphicPack/GraphicPack2PatchesApply.cpp:135]: Ripping apart the two similar `.substr()` lines worsens readability imho. I assume this was done as an optimization but the else case is taken virtually always as it's the non-error case.

File: src/Cafe/HW/Latte/Core/LatteRenderTarget.cpp [hunk 3]
```diff
@@ -486,38 +486,41 @@ bool LatteMRT::UpdateCurrentFBO()
...
-		// set effective size
-		sint32 effectiveWidth, effectiveHeight;
-		LatteTexture_getEffectiveSize(colorAttachmentView->baseTexture, &effectiveWidth, &effectiveHeight, nullptr, colorAttachmentView->firstMip);
-		if( rtEffectiveSize->width == 0 && rtEffectiveSize->height == 0 )
-		{
-			rtEffectiveSize->width = effectiveWidth;
-			rtEffectiveSize->height = effectiveHeight;
-		}
-		else if( rtEffectiveSize->width != effectiveWidth && rtEffectiveSize->height != effectiveHeight )
+		if (colorAttachmentView)
```
Comment (Exzap) [inline src/Cafe/HW/Latte/Core/LatteRenderTarget.cpp:489]: Unnecessary null pointer check. GetColorAttachmentTexture() will never return null.

File: src/Cafe/HW/Latte/LegacyShaderDecompiler/LatteDecompilerAnalyzer.cpp [hunk 4]
```diff
@@ -850,9 +850,12 @@ void LatteDecompiler_analyze(LatteDecompilerShaderContext* shaderContext, LatteD
 			cemu_assert_unimplemented();
 	}
 	// analyze subroutines
-	for (auto subroutineAddr : list_subroutineAddrs)
+	if (!list_subroutineAddrs.empty())
```
Comment (Exzap) [inline src/Cafe/HW/Latte/LegacyShaderDecompiler/LatteDecompilerAnalyzer.cpp:853]: What is the advantage of checking if non-empty for a range based loop?

File: src/Cafe/OS/libs/vpad/vpad.cpp [hunk 5]
```diff
@@ -905,11 +905,11 @@ void vpadExport_VPADStopMotor(PPCInterpreter_t* hCPU)
 void vpadExport_VPADSetTPCalibrationParam(PPCInterpreter_t* hCPU)
 {
 	ppcDefineParamU32(channel, 0);
-	ppcDefineParamMEMPTR(params, VPADTPCalibrationParam, 1);
 
 	debugBreakpoint();
 	if (channel < VPAD_MAX_CONTROLLERS)
 	{
+		ppcDefineParamMEMPTR(params, VPADTPCalibrationParam, 1);
```
Comment (Exzap) [inline src/Cafe/OS/libs/vpad/vpad.cpp:912]: Parameter initialization should be at the beginning of the function. There is also no performance gained as this code path is taken 99.99% of time.

File: src/Cemu/FileCache/FileCache.cpp [hunk 6]
```diff
@@ -287,7 +287,10 @@ uint8* _fileCache_compressFileData(const uint8* fileData, uint32 fileSize, sint3
 	Bytef* compressedData = (Bytef*)malloc(4 + compressedLen);
 	int zret = compress2(compressedData + 4, &compressedLen, uncompressedInput, uncompressedLen, 4); // level 4 has good compression to performance ratio
 	if (zret != Z_OK)
+	{
+		delete compressedData;
```
Comment (Exzap) [inline src/Cemu/FileCache/FileCache.cpp:291]:
```suggestion
		free(compressedData);
```
Delete is wrong here as the memory is allocated with `malloc`

File: src/Cemu/nex/nexFriends.h [hunk 7]
```diff
@@ -175,10 +172,10 @@ class nexPrincipalBasicInfo : public nexType
 	}
 
 	nexPrincipalBasicInfo(uint32 principalId, char* nnid, const nexMiiV2& mii)
+		: mii(mii)
```
Comment (Exzap) [inline src/Cemu/nex/nexFriends.h:175]: `principalId` should also be moved into the initializer list

File: src/Cemu/nex/nexFriends.h [hunk 8]
```diff
@@ -218,10 +215,10 @@ class nexNNAInfo : public nexType
 	}
 
 	nexNNAInfo(uint8 countryCode, uint8 countrySubCode, const nexPrincipalBasicInfo& principalBasicInfo)
+		: principalInfo(principalBasicInfo)
```
Comment (Exzap) [inline src/Cemu/nex/nexFriends.h:218]: `countryCode` and `countrySubCode` should also be part of the initializer list

File: src/input/InputManager.cpp [hunk 9]
```diff
@@ -225,14 +225,14 @@ bool InputManager::migrate_config(const fs::path& file_path)
...
 		const auto api_string = m_data.get<std::string>("General.api");
-		auto uuid_opt = m_data.get_optional<std::string>("General.controller");
 		const auto display_name = m_data.get_optional<std::string>("General.display");
 
 		std::string uuid;
 		if (api_string == to_string(InputAPI::Keyboard))
 			uuid = to_string(InputAPI::Keyboard);
 		else
 		{
+			auto uuid_opt = m_data.get_optional<std::string>("General.controller");
```
Comment (Exzap) [inline src/input/InputManager.cpp:235]: Seems like an unnecessary micro-optimization that only sacrifices readability?

**Review by Exzap — CHANGES_REQUESTED** [review]
Overall this is good. Some of it is pretty subjective but still acceptable. However, a few of the proposed changes are micro-optimizations at the cost of code quality which need to be addressed.

File: src/Cafe/HW/Latte/LegacyShaderDecompiler/LatteDecompilerAnalyzer.cpp (hunk 4 above)
Comment (GermanAizek) [reply src/Cafe/HW/Latte/LegacyShaderDecompiler/LatteDecompilerAnalyzer.cpp:853]: Yes, you're right, after C++11, don't need write this
Reference: https://stackoverflow.com/questions/52828423/in-c11-and-later-is-it-still-optimal-to-do-an-empty-check-before-a-range-f

**Comment by GermanAizek** [comment]
@Exzap, fixed your changes here https://github.com/cemu-project/Cemu/pull/258/commits/89c0945f23a31f2a55bae75f22f03d49cba7d785

**Comment by Exzap** [comment]
Looks good. Thanks!

**Pull request merged**